generalize the Action

adaliszk opened the pull request:
+ Changes from @AndreLobato to allow the Action to be used with any registry
+ Make registry optional so it could fallback to the Runner default
+ Add Env variable fallback, so the Action may be parameterized that way
+ Add some Logging to the output so it would show what's getting exected
+ Update to the README about the usage

koraykoska commented on .gitignore:
```diff
@@ -1 +1,2 @@
 .DS_Store
+.idea
```
Newline at the end please

koraykoska commented on entrypoint.sh:
```diff
@@ -1,24 +1,48 @@
+    fi
 }
+
+main
```
Also here the newline.

koraykoska commented on action.yml:
```diff
@@ -1,29 +1,33 @@
-    description: 'The name of the image you would like to retag'
-    required: true
+    description: 'The name of the image you would like to re-tag'
+    required: false
```
Why is the name optional now? And the others too.

adaliszk replied:
I wanted to use `env` variables as alternatives to `with` and when they are required you cannot ignore them in the workflow. I guess `default` could be something to add here, but even then I am unsure if the users needs to still specify a `with` section or not.

adaliszk replied:
Added required back with defaults 😉  it works just fine

koraykoska reviewed (commented):
Just let me know a little more information on why those parameters are optional now.

adaliszk commented:
Added required back as `default` will fill it out just fine, though I added a 2nd layer in the script to test if there is a value.
Additionally, I've added a simple workflow to test the action with code changes.

koraykoska commented:
Thanks @adaliszk 
Please bump to 0.2.0 as I see some breaking changes. As long as we are in 0.x.x we can do minor bumps for breaking changes.
I will merge it once you pushed the 0.2.0. Let me know so I get notified.
